src/device/mod.rs

```diff
@@ -1436,6 +1448,20 @@ impl Runtime {
         Ok(())
     }
 
+    async fn notify_sleep(&mut self) -> Result {
+        if let Some(nurse) = &self.entities.nurse {
+            nurse.handle_sleep().await;
+        }
```

Comment:
Events from WG should take care of that IMO.
@Jauler  what is your opinion ?

Reply:
Generally all data is affected by sleeps, and we do have some incorrect reporting for older Quality Of Service metrics. But I believe it will be more efficient to address it with channel refactoring in LLT-5018 instead if using this PR.